src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -228,6 +249,21 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseSer
             complaints = complaintService.getAllComplaintsByExerciseIdAndTutorId(exerciseId, tutorId, isAtLeastInstructor);
         }
 
-        return ResponseEntity.ok(complaints);
+        return ResponseEntity.ok(getComplaintsByComplaintType(complaints, complaintType));
+    }
+
+    /**
+     * Filter out all complaints that are not of a specified type. Since enum can't be null and we have only two enum values, we filter always by more feedback value
+     *
+     * @param complaints    list of complaints
+     * @param complaintType the type of complaints we want to get
+     */
+    private List<Complaint> getComplaintsByComplaintType(List<Complaint> complaints, ComplaintType complaintType) {
+        if (complaintType == ComplaintType.MORE_FEEDBACK) {
+            return complaints.stream().filter(complaint -> complaint.getComplaintType() == ComplaintType.MORE_FEEDBACK).collect(Collectors.toList());
+        }
+        else {
```

Comment:
What happens if we have more complaint types in the future? I suggest to use a switch case statement here and throw and leave out the default case. Then we are notified by a compiler error.

Alternatively you could simply pass the complaintType instead of checking it.

(Please do this in other places as well).

Reply:
When we introduce new column complaint type is it possible to set all existing complaints to `Complaint` type? If it is then I don't have to check for null values here.

Reply:
I resolved it now by adding not null constraint right after I add new column `complaintType`. I removed all the null checks from code since complaint should always have a `complaintType`